modules/installation-extend-edge-nodes-aws-local-zones.adoc

```diff
@@ -0,0 +1,106 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/cluster-tasks.adoc
+
+ifeval::["{context}" == "installing-aws-localzone"]
+:localzone:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="installation-extend-edge-nodes-aws-local-zones_{context}"]
+= Creating user workloads in AWS Local Zones
+After you create an Amazon Web Service (AWS) Local Zone environment, and you deploy your cluster, you can use edge worker nodes to create a user workloads in Local Zone subnets. 
+
+After the `openshift-installer` creates the cluster, the installer automatically specifies a taint effect of `NoSchedule` to each edge worker node. This means that a scheduler does not add a new pod, or deployment, to a node if the pod does not match the specified tolerations for a taint. You can modify the taint for better control over how each node creates a workload in each Local Zone subnet.
+
+The `openshift-installer` creates the machine set manifests file with `node-role.kubernetes.io/edge` and `node-role.kubernetes.io/worker` labels applied to each edge worker node that is located in a Local Zone subnet.
+
+.Prerequisites
+
+* You have an access to the OpenShift CLI (`oc`).
+* You deployed your cluster in a Virtual Private Cloud (VPC) with defined Local Zone subnets.
+* You ensured that the machine set for the edge workers on Local Zone subnets specifies the taints for `node-role.kubernetes.io/edge`.
+
+.Procedure
+
+. Create a `deployment` resource YAML file for an example application to be deployed in the edge worker node that operates in a Local Zone subnet. Ensure that you specify the correct tolerations that match the taints for the edge worker node.
++
+.Example of a configured `deployment` resource for an edge worker node that operates in a Local Zone subnet
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment <1>
+metadata:
+  name: <local_zone_application> <2>
+  namespace: <local_zone_application_namespace> <3>
+spec:
+  selector: 
+    matchLabels:
+      app: <local_zone_application>
+  replicas: 1
+  template:
+    metadata:
+      labels: 
+        app: <local_zone_application>
+        machine.openshift.io/zone-group: ${ZONE_GROUP_NAME} <4>
+    spec:
+      nodeSelector: <5>
+        machine.openshift.io/zone-group: ${ZONE_GROUP_NAME}
+      tolerations: <6>
+      - key: "node-role.kubernetes.io/edge"
+        operator: "Equal"
+        value: ""
+        effect: "NoSchedule"
+      containers:
+        - image: openshift/origin-node
+          command:
+           - "/bin/socat"
+          args:
+            - TCP4-LISTEN:8080,reuseaddr,fork
+            - EXEC:'/bin/bash -c \"printf \\\"HTTP/1.0 200 OK\r\n\r\n\\\"; sed -e \\\"/^\r/q\\\"\"'
+          imagePullPolicy: Always
+          name: echoserver
+          ports:
+            - containerPort: 8080
+    volumes:
+    - name: data 
+      ephemeral:
```

Comment:
This example was based on ephemeral volumes[1]. I don't have a strong opinion on keeping the ephemeral or regular as an example. WDYT  @yunjiang29 ?

[1] https://docs.openshift.com/container-platform/4.12/storage/generic-ephemeral-vols.html

Reply:
Hi @dfitzmau @mtulio , agree that the ephemeral volume is not a good example as the volume will be cleaned up when the deployment deleted. It's not a good example for user general usage scenarios (persistent volume). A general example should be using workload with pvcs. We could replace the data volume `volume[0]` with persistent volume.
E.g.
```console
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: mypvc
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: gp2-csi
  volumeMode: Filesystem
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: mydeployment
spec:
  selector:
    matchLabels:
      app: myapp
  replicas: 1
  template:
    metadata:
      labels:
        app: myapp
    spec:
      containers:
      - name: myc1
        image: quay.io/pewang/storagetest:base
        volumeMounts:
        - mountPath: /mnt/storage
          name: data
      volumes:
      - name: data
        persistentVolumeClaim:
          claimName: mypvc

```

Reply:
Thanks, @Phaow and @mtulio . I updated the example resource. I retained the `spec` stanza, because it details tolerations and so on. WDYT?

Reply:
looks good to me. Thanks for the update!

Reply:
Hi @mtulio . Are you OK with this spec update? I can then send this PR onto the next review stage.

Reply:
@dfitzmau for sure, I am ok (ty @Phaow ).

I would like to validate the entire flow in the p1 and p2 clusters, but it will not block you to move on to the next review stage.